src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -829,6 +829,28 @@ namespace smt::noodler {
         // Add dummy symbols for all disequations.
         // FIXME: we can possibly create more dummy symbols than the size of alphabet (196607 - from string theory standard), but it is edge-case that is nearly impossible to happen
         std::set<uint32_t> dummy_symbols{ util::get_dummy_symbols(std::max(new_symbs, size_t(3)), symbols_in_formula) };
+
+        // satisfiability checking via universality checking
```

Comment:
Can you explain this better? That we are doing universality checking as an optimization in the case that we have only one constraint that is x \not\in RL (if I understand correctly).